### ISSUE TITLE
YOLO11 with ReLU converges poorly compared to YOLOv8 after replacing SiLU (INT8 quantization use-case)

### ISSUE THREAD

**Issue by sihogu** [ultralytics/ultralytics#23993]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am currently training YOLO11 for an edge device deployment.

For INT8 quantization constraints on my device, I replaced the activation functions in the convolution blocks from SiLU to ReLU and retrained the models.

However, I encountered an issue:
When both YOLOv8 and YOLO11 are trained with ReLU activation, YOLOv8 achieves better performance than YOLO11.

Specifically:

- YOLO11 training seems to converge too early, possibly getting stuck in a local minimum
- YOLOv8 continues to train more stably and achieves better results

My current hypothesis is:

- YOLOv8, having relatively larger parameter capacity, is more robust to the loss of representational power caused by replacing SiLU with ReLU

- YOLO11, being lighter, suffers more from reduced expressiveness after removing SiLU, which negatively impacts convergence

In my case, improving the performance of ReLU-based YOLO11 is the main goal.

So I would like to ask:

1. Is my hypothesis reasonable?
2. Could there be other reasons why YOLO11 does not converge well with ReLU?
3. Do you have any recommendations to improve training performance of YOLO11 under ReLU (especially for INT8 deployment scenarios)?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sihogu, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this looks like a custom training ❓ question, please provide as much information as possible, including dataset image examples, training logs, your exact model/config changes, and confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
If you believe there may also be a training regression or implementation issue involved, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate it efficiently 🔍
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you post the mAP results with each training?

**Comment by sihogu** [comment]
@Y-T-G

The training was conducted using the small model with 1600 epochs and a patience of 50. Since a YAML-defined model was used, the training was performed from scratch.

The FP32 GPU results are as follows:

- YOLO11:

Epochs: 756
mAP50: 0.579
mAP50-95: 0.418

- YOLOv8:

Epochs: 1600
mAP50: 0.613
mAP50-95: 0.445

**Comment by Y-T-G** [comment]
How many images do you have?

**Comment by sihogu** [comment]
This is the default COCO dataset.

**Comment by glenn-jocher** [comment]
Thanks, that helps—on COCO from scratch this looks more like an activation/architecture effect than a dataset-size issue. Your hypothesis is reasonable: `YOLO11`’s `C3k2`/`C2PSA` design and the default `SiLU`-based conv blocks appear more sensitive to a global `ReLU` swap than `YOLOv8`’s `C2f` design, so for INT8 I’d first compare against standard `YOLO11` with backend quantization, and if `ReLU` is mandatory try a larger `YOLO11` variant or `ReLU6` with a fresh tune rather than expecting parity from a drop-in activation change ([YOLO11 vs YOLOv8](https://docs.ultralytics.com/compare/yolo11-vs-yolov8/), [conv module reference](https://docs.ultralytics.com/reference/nn/modules/conv/)).

**Comment by sihogu** [comment]
@glenn-jocher 
Thank you for your helpful response.

To clarify, all the results I mentioned earlier are based on FP32 training and evaluation.
After applying INT8 PTQ and evaluating on our device, we observed a severe performance drop when using the SiLU-based model—the score drops to around 1. 

This seems to be due to our device’s quantization constraint, where the zero-point is fixed at 0, causing large quantization errors for negative values produced by SiLU.

For this reason, we decided to move away from SiLU and retrained the models using ReLU instead. Our current goal is to improve the FP32 performance of YOLO11 with ReLU.

Thanks again for your insights. If you have any additional suggestions regarding this situation, I would greatly appreciate it. Otherwise, I think this will conclude my questions.

**Comment by glenn-jocher** [comment]
Given that fixed-zero-point INT8 is the real constraint, I’d retune `YOLO11` from scratch with a lower initial LR, longer warmup, and no early stopping, and if your backend supports it prioritize quantization-aware training over PTQ since that usually matters more than maximizing FP32 after a global `ReLU` swap; this [pruning and quantization guide](https://www.ultralytics.com/blog/pruning-and-quantization-in-computer-vision-a-quick-guide) may help frame the tradeoffs.

**Comment by sihogu** [comment]
Thank you very much for your valuable advice. I will proceed with further experiments based on your suggestions.

**Comment by glenn-jocher** [comment]
Sounds good, thanks for following up—if you get a reproducible `YOLO11` + `ReLU` setup that still plateaus after retuning, please share the exact model YAML, training args, full metrics, and `yolo checks` output here and we can take another look.
